search/api.py

```diff
@@ -1,306 +1,143 @@
-            'graduation_date': DATE_TYPE,
-            'id': LONG_TYPE,
-            'online_degree': BOOL_TYPE,
-            'school_city': NOT_ANALYZED_STRING_TYPE,
-            'school_country': NOT_ANALYZED_STRING_TYPE,
-            'school_name': NOT_ANALYZED_STRING_TYPE,
-            'school_state_or_territory': NOT_ANALYZED_STRING_TYPE,
-        }},
-        'email_optin': BOOL_TYPE,
-        'filled_out': BOOL_TYPE,
-        'first_name': NOT_ANALYZED_STRING_TYPE,
-        'gender': NOT_ANALYZED_STRING_TYPE,
-        'has_profile_image': BOOL_TYPE,
-        'last_name': NOT_ANALYZED_STRING_TYPE,
-        'preferred_language': NOT_ANALYZED_STRING_TYPE,
-        'preferred_name': NOT_ANALYZED_STRING_TYPE,
-        'pretty_printed_student_id': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_full': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_large': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_medium': NOT_ANALYZED_STRING_TYPE,
-        'profile_url_small': NOT_ANALYZED_STRING_TYPE,
-        'username': NOT_ANALYZED_STRING_TYPE,
-        'work_history': {'type': 'nested', 'properties': {
-            'city': NOT_ANALYZED_STRING_TYPE,
-            'company_name': NOT_ANALYZED_STRING_TYPE,
-            'country': NOT_ANALYZED_STRING_TYPE,
-            'id': LONG_TYPE,
-            'industry': NOT_ANALYZED_STRING_TYPE,
-            'position': NOT_ANALYZED_STRING_TYPE,
-            'start_date': DATE_TYPE,
-            'state_or_territory': NOT_ANALYZED_STRING_TYPE,
-        }},
-    })
-    mapping.field("program", "object", properties={
-        'id': LONG_TYPE,
-        'grade_average': LONG_TYPE,
-        'enrollments': {'type': 'nested', 'properties': {
-            'course_details': {
-                'type': 'object',
-                'properties': {
-                    'course_modes': {
-                        'type': 'nested',
-                    }
-                }
-            }
-        }},
-        'certificates': {'type': 'nested'}
-    })
-    # Make strings not_analyzed by default
-    mapping.meta('dynamic_templates', [{
-        "notanalyzed": {
-            "match": "*",
-            "match_mapping_type": "string",
-            "mapping": NOT_ANALYZED_STRING_TYPE
-        }
-    }])
-    return mapping
+    __call__ = execute
 
 
-def create_program_enrolled_user_mapping():
+class ESFullResultEmailSearch(ESFullResultSearch):
     """
-    Save the mapping for a program user. If one already exists, delete it first.
+    Provides a function to return all unique emails for documents that match
+    an ES query
     """
-    conn = get_conn(verify=False)
-    index_name = settings.ELASTICSEARCH_INDEX
-    if conn.indices.exists_type(index=index_name, doc_type=USER_DOC_TYPE):
-        conn.indices.delete_mapping(index=index_name, doc_type=USER_DOC_TYPE)
-    mapping = program_enrolled_user_mapping()
-    mapping.save(index_name)
+    def init_combined_results(self):
```

Comment:
Can you change this and `add_to_results` to have an underscore before them if they're not meant to be used outside the class?